datafusion/src/optimizer/simplify_expressions.rs

```diff
@@ -900,28 +809,99 @@ mod tests {
         );
         let expected = expr.clone();
 
-        assert_eq!(simplify(&expr), expected);
+        assert_eq!(simplify(expr), expected);
     }
 
     #[test]
     fn test_simplify_or_and() {
-        // (c > 5) OR ((d < 6) AND (c > 5) -- can remove
-        let expr = binary_expr(
-            col("c2").gt(lit(5)),
+        let l = col("c2").gt(lit(5));
```

Comment:
additional coverage showing that this rewrite is not possible unless you know `c1 < 6` can not be null